src/Microsoft.Restier.EntityFramework7/Submit/ChangeSetPreparer.cs

```diff
@@ -116,78 +126,110 @@ private ChangeSetPreparer()
             return entity;
         }
 
-        private static void SetValues(EntityEntry dbEntry, DataModificationEntry entry, Type entityType)
+        private static object CreateFullUpdateInstance(DataModificationEntry entry, Type entityType)
         {
-            //StateEntry stateEntry = ((IAccessor<InternalEntityEntry>) dbEntry.StateEntry;
-            IEntityType edmType = dbEntry.Metadata;
+            // The algorithm for a "FullReplaceUpdate" is taken from ObjectContextServiceProvider.ResetResource
+            // in WCF DS, and works as follows:
+            //  - Create a new, blank instance of the entity.
+            //  - Copy over the key values and set any updated values from the client on the new instance.
+            //  - Then apply all the properties of the new instance to the instance to be updated.
+            //    This will set any unspecified properties to their default value.
+            object newInstance = Activator.CreateInstance(entityType);
+
+            ChangeSetPreparer.SetValues(newInstance, entityType, entry.EntityKey);
+            ChangeSetPreparer.SetValues(newInstance, entityType, entry.LocalValues);
+
+            return newInstance;
+        }
 
-            if (entry.IsFullReplaceUpdate)
+        private static void SetValues(EntityEntry dbEntry, DataModificationEntry entry, Type entityType)
+        {
+            foreach (KeyValuePair<string, object> propertyPair in entry.LocalValues)
             {
-                // The algorithm for a "FullReplaceUpdate" is taken from WCF DS ObjectContextServiceProvider.ResetResource, and is as follows:
-                // Create a new, blank instance of the entity.  Copy over the key values, and set any updated values from the client on the new instance.
-                // Then apply all the properties of the new instance to the instance to be updated.  This will set any unspecified
-                // properties to their default value.
+                PropertyEntry propertyEntry = dbEntry.Property(propertyPair.Key);
+                Type type = propertyEntry.CurrentValue.GetType();
```

Comment:
Actually here is a bug because the current value could be null thus there is no type here. I've fixed the bug in EF6 project but not EF7. I think we shall have a task #280 to sync the latest features from EF6 to EF7.